src/containers/guest-home-page/google-login/GoogleLogin.js

```diff
@@ -1,11 +1,24 @@
+import { useContext } from 'react'
 import { useTranslation } from 'react-i18next'
 import { Box, Typography, Button } from '@mui/material'
+import { Link } from 'react-router-dom'
+
+import { ModalContext } from '~/context/modal-context'
+import { routes } from '~/constants/routes'
+import LoginDialog from '~/containers/guest-home-page/login-dialog/LoginDialog'
 
 import { style } from '~/containers/guest-home-page/google-login/google-login.style'
 import google from '~/assets/img/login-dialog/google.svg'
 
 const GoogleLogin = ({ type }) => {
   const { t } = useTranslation()
+  const { whatCanYouDo } = routes.guestNavBar
+  const { setModal, closeModal } = useContext(ModalContext)
+
+  const openLoginDialog = () => {
+    closeModal()
```

Comment:
can't we just setModal?

Reply:
yeah but at first transition there is a twitch a bit